bedwars-plugin/src/main/java/com/andrei1058/bedwars/arena/upgrades/HealPoolTask.java

```diff
@@ -87,7 +87,8 @@ public static boolean exists(IArena arena, ITeam bwt){
     }
     public static void removeForArena(IArena a){
         if (healPoolTasks.isEmpty() || a == null) return;
-        for (HealPoolTask hpt: healPoolTasks) {
+        for (int i = 0; i < healPoolTasks.size(); i++) {
+            HealPoolTask hpt = healPoolTasks.get(i);
             if (hpt == null) continue;
             if (hpt.getArena().equals(a)){
                 hpt.cancel();
```

Comment:
Please check [the answer](https://stackoverflow.com/a/6046087). Iterating forward while removing items is not correct. Using the accepted answer in the link or `removeIf` would be a much better solution.

```java
healPoolTasks.removeIf(hpt -> {
    if (hpt != null && hpt.arena.equals(a)) {
        hpt.cancel();
        return true;
    }
    return false;
});
```